src/aspire/utils/misc.py

```diff
@@ -439,6 +440,30 @@ def cyclic_rotations(order, dtype=np.float64):
     return rots_symm
 
 
+def check_pixel_size(pixel_sizes, reference_pixel_size):
+    """
+    Ensure that `reference_pixe_size` is a scalar and check that `pixel_sizes`
+    are close to `reference_pixel_size`, warn if not.
+
+    :param pixel_sizes: Pixel size value(s) to check (scalar or ndarray).
+    :param reference_pixel_size: Reference value (scalar).
+    :return: True if close, False otherwise.
+    """
+    if not np.asarray(reference_pixel_size).size == 1:
+        raise ValueError("`reference_pixel_size` must be a scalar.")
+
+    close = np.allclose(pixel_sizes, reference_pixel_size)
+    if not close:
+        warnings.warn(
+            f"User provided pixel_size: {reference_pixel_size} angstrom, does not match"
+            f" pixel_size found in metadata: {pixel_sizes} angstrom. Setting"
```

Comment:
This is a little confusing, since I don't think we always check the return value and act on it. Should this be updated?